beluga/test/beluga/views/test_take_evenly.cpp

```diff
@@ -0,0 +1,89 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gmock/gmock.h>
+
+#include "beluga/views/take_evenly.hpp"
+
+#include <range/v3/range/conversion.hpp>
+
+namespace {
+
+TEST(TakeEvenlyView, NoElementsTakeZero) {
+  const auto input = std::vector<int>{};
+  const auto output = input | beluga::views::take_evenly(0) | ranges::to<std::vector>;
+  ASSERT_EQ(output.size(), 0);
+}
+
+TEST(TakeEvenlyView, NoElements) {
+  const auto input = std::vector<int>{};
+  const auto output = input | beluga::views::take_evenly(1) | ranges::to<std::vector>;
+  ASSERT_EQ(output.size(), 0);
+}
+
+TEST(TakeEvenlyView, TakeZero) {
+  const auto input = std::vector<int>{1, 2, 3, 4};
+  const auto output = input | beluga::views::take_evenly(0) | ranges::to<std::vector>;
+  ASSERT_EQ(output.size(), 0);
+}
+
+TEST(TakeEvenlyView, TakeOne) {
+  const auto input = std::vector<int>{1, 2, 3, 4};
+  const auto output = input | beluga::views::take_evenly(1) | ranges::to<std::vector>;
+  ASSERT_THAT(output, testing::ElementsAre(1));
+}
+
+TEST(TakeEvenlyView, TakeAll) {
+  const auto input = std::vector<int>{1, 2, 3, 4};
+  const auto output = input | beluga::views::take_evenly(10) | ranges::to<std::vector>;
+  ASSERT_THAT(output, testing::ElementsAre(1, 2, 3, 4));
+}
+
+TEST(TakeEvenlyView, TakeTwoFromFour) {
+  const auto input = std::vector<int>{1, 2, 3, 4};
+  const auto output = input | beluga::views::take_evenly(2) | ranges::to<std::vector>;
+  ASSERT_THAT(output, testing::ElementsAre(1, 4));
+}
+
+TEST(TakeEvenlyView, TakeThreeFromFive) {
+  const auto input = std::vector<int>{1, 2, 3, 4, 5};
+  const auto output = input | beluga::views::take_evenly(3) | ranges::to<std::vector>;
+  ASSERT_THAT(output, testing::ElementsAre(1, 3, 5));
+}
+
+TEST(TakeEvenlyView, TakeThreeFromSix) {
+  const auto input = std::vector<int>{1, 2, 3, 4, 5, 6};
+  const auto output = input | beluga::views::take_evenly(3) | ranges::to<std::vector>;
+  ASSERT_THAT(output, testing::ElementsAre(1, 4, 6));
+}
+
+TEST(TakeEvenlyView, TakeThreeFromNine) {
+  const auto input = std::vector<int>{1, 2, 3, 4, 5, 6, 7, 8, 9};
+  const auto output = input | beluga::views::take_evenly(3) | ranges::to<std::vector>;
+  ASSERT_THAT(output, testing::ElementsAre(1, 5, 9));
+}
+
+TEST(TakeEvenlyView, TakeThreeFromFour) {
+  const auto input = std::vector<int>{1, 2, 3, 4};
+  const auto output = input | beluga::views::take_evenly(3) | ranges::to<std::vector>;
+  ASSERT_THAT(output, testing::ElementsAre(1, 3, 4));
+}
+
+TEST(TakeEvenlyView, TakeFiveFromNine) {
```

Comment:
this is on me: the name of the test is not correct.

Reply:
Good catch! Fixed in https://github.com/Ekumen-OS/beluga/pull/271/commits/395d0ee2f9784cdcf9ae2df21b4fa371545731f0.